server/shared/src/test/scala/org/http4s/server/middleware/ThrottleSuite.scala

```diff
@@ -66,21 +66,10 @@ class ThrottleSuite extends Http4sSuite {
 
       val takeTokenAfterRefill = createBucket.flatMap { testee =>
         testee.takeToken *> IO.sleep(exceeded) *>
-          testee.takeToken
+          (IO.realTime, testee.takeToken).mapN((_, _))
```

Comment:
Nice! But we should swap the order here, so that the timestamp is recorded after `takeToken` completes :)

Reply:
Or maybe use `parMapN` and record it in parallel? All in all here we're already past the point of the refill time

Reply:
> All in all here we're already past the point of the refill time.

Ah, that's a fair point. I'm actually not sure what the value of checking the timestamp here is? Given that `testee.takeToken` is non-blocking, it seems to me it is testing if `IO.sleep` works :)

Reply:
Yeah, I'm actually thinking: here we want to test that a particular function return some result only *after* some threshold, hence what we should do is:
1. test that before that threshold the outcome is not the expected value
2. test that after instead it is
So, that's what I'm going to implement - perhaps this was the missing bit in the first iteration!